.github/workflows/build.yml

```diff
@@ -40,11 +39,48 @@ jobs:
       - name: docker login
         uses: aws-actions/amazon-ecr-login@v1
 
+      # for staging
+      # gather metadata from git & github actions to reference in docker
+      - name: git & github metadata staging
+        id: metadata_staging
+        uses: docker/metadata-action@v3
+        if: ${{ github.event.inputs.is_production == false }}
+        with:
+          images: ${{ env.AWS_ACCOUNT_ID }}.dkr.ecr.${{ env.AWS_REGION }}.amazonaws.com/openverse/frontend
+          tags: |
+            latest=auto
+            type=edge,branch=main
+
       # build a docker image
       - name: build docker image
         uses: docker/build-push-action@v2
+        if: ${{ github.event.inputs.is_production == false }}
         with:
           context: .
-          tags: ${{ steps.metadata.outputs.tags }}
-          labels: ${{ steps.metadata.outputs.labels }}
+          tags: ${{ steps.metadata_staging.outputs.tags }}
+          labels: ${{ steps.metadata_staging.outputs.labels }}
           push: true
+
+      # for production
+      # gather metadata from git & github actions to reference in docker
+      - name: git & github metadata production
```

Comment:
There are a lot of similarities between this and the staging step. I wonder if we might be able to combine the logic and add conditions in certain places for each deployment case.

Reply:
That's a good point, and that's what I [mentioned](https://a8c.slack.com/archives/C029NN9G24C/p1640635252388500) when I officially announced this PR.

This is something that will be improved in the future based on what we are going to see when deploying this on the real infrastructure.

By now, I'm separating the logic between staging and production, but I'm planning to send a PR to the [official action project (owned by Docker)](https://github.com/docker/metadata-action) that allows me to manipulate the tags without the need to duplicate the whole step.

cc @zackkrida